### ISSUE TITLE
JOSS Review

### ISSUE THREAD

**Issue by appelmar** [r-spatial/rgee#70]
Great work @csaybar! Here is my review of the JOSS submission (openjournals/joss-reviews#2272).

# General Comments

The rgee package is a welcome addition to the R spatial ecosystem. It allows to combine the processing of massive Earth observation datasets in Google Earth Engine (GEE) with R-based analysis workflows (e.g., to use R for pre/post-processing, visualization, and mapping). The amount of examples available online (https://csaybar.github.io/rgee-examples/) is impressive and very helpful for users to familiarize with GEE / rgee concepts. 

All in all, the package and the paper are well written and the documentation is very comprehensive. The package clearly makes it *very easy* to use GEE from R. Below, please find some minor(!) comments for the paper and the software separately.

# Paper

The paper highlights most important design decisions of the package and includes helpful code examples. I only have two minor comments:

- The source code comment in the time series extraction example _"Image or ImageCollection (mean composite)"_ is not so clear, I'd suggest to describe what you actually do in the lines below (select dataset, reproject, filter by date and variable).

- The paper mentions that rgee is available from CRAN but it is not (yet).

# Software

- Installation (including a Miniconda Python environment) went smooth on my platform (Ubuntu 18.04, R 4.0.1, and rgee package version 0.6.2, commit 454eb81102802813516ce7fb21986737e37e5581). 

- For the `region` argument in  `ee_as_stars()` and `ee_as_raster()`, the documentation states that "if not specified, image bounds will be taken". However, in my tests, I get _"Error in ee_image_local(image = image, region = region, dsn = dsn, via = via,  : argument "region" is missing, with no default"_ when calling these functions without region argument.

- I've found that calling `ee_Initialize(drive = TRUE, gcs = TRUE)` sometimes constantly asks for a selection of the mail address (though there is only one listed). If selecting 1, it asks again whereas if selecting 0, the package opens the browser for authorization, and after allowing everything, it asks again. Am I doing something wrong? I don't know any Google API details but if possible, I think it would be nice to make the Google drive / cloud storage authorization as simple as when calling `ee_Initialize()` for GEE only (which works well).

- It is clear that function examples in the documentation are marked as `dontrun`, because they need a GEE account. However, I would have enjoyed to run R CMD check with argument `--run-dontrun` with my own GEE acount, and also run the testthat tests. I was not successful because it seems that the GEE user is hard-coded in the tests and some examples (with Google drive or cloud storage) do not work in a non-interactive session. Do you see any way to make it easier for others to run all tests / examples automatically?

- `R CMD check --as-cran` reports some documentation issues of `print.ee.computedobject.ComputedObject()`: _"Found the following apparent S3 methods exported but not registered: print.ee.computedobject.ComputedObject"_

- Nice work on the `ee_help()` function! It is very helpful to have the GEE documentation of functions in the R help viewer.

**Comment by csaybar** [comment]
Hi @appelmar  thank you so much for taking your time in review rgee. I implemented the changes in #69 (See [NEWS.md](https://github.com/r-spatial/rgee/blob/JOSS/NEWS.md)) and recompiled the paper (see https://github.com/r-spatial/rgee/blob/JOSS/paper/paper.md). I commented on all your suggestions. Please let me know if there are any issues remaining. I will keep #70 open until all the review process was completed.

Have a nice day!

> The source code comment in the time series extraction example "Image or ImageCollection (mean composite)" is not so clear, I'd suggest to describe what you actually do in the lines below (select dataset, reproject, filter by date and variable).

You're right, I made changes in the paper to solve this issue. 

> The paper mentions that rgee is available from CRAN but it is not (yet).

Although I made the changes before. It seems I not compiled the paper correctly. I remove this part of the paper.
We have some problems to publish the package in CRAN (It was because rgee did not respect all CRAN policy since many functions used to manipulate the .Renviron without consulting users). We decided to wait to have the JOSS decision before resubmitting the package.

> For the region argument in ee_as_stars() and ee_as_raster(), the documentation states that "if not specified, image bounds will be taken". However, in my tests, I get "Error in ee_image_local(image = image, region = region, dsn = dsn, via = via, : argument "region" is missing, with no default" when calling these functions without region argument.

Thanks, we fix this, now the following code will work:

```r
library(rgee)
library(stars)
ee_Initialize()

ee$Image("LANDSAT/LC08/C01/T1_SR/LC08_038029_20180810") %>% 
  ee$Image$select(c("B4", "B3", "B2")) %>% 
  ee$Image$divide(10000) %>% 
  ee_as_stars(scale = 1000) -> l8_img
plot(l8_img, rgb = c(1,2,3))
```
>  I've found that calling ee_Initialize(drive = TRUE, gcs = TRUE) sometimes constantly asks for a selection of the mail address (though there is only one listed). If selecting 1, it asks again whereas if selecting 0, the package opens the browser for authorization, and after allowing everything, it asks again. Am I doing something wrong? I don't know any Google API details but if possible, I think it would be nice to make the Google drive / cloud storage authorization as simple as when calling ee_Initialize() for GEE only (which works well).

This was a big mistake. Thank you very much for pointing it out. It was fixed, please let us know if the problem persists.

> It is clear that function examples in the documentation are marked as dontrun, because they need a GEE account. However, I would have enjoyed to run R CMD check with argument --run-dontrun with my own GEE acount, and also run the testthat tests. I was not successful because it seems that the GEE user is hard-coded in the tests and some examples (with Google drive or cloud storage) do not work in a non-interactive session. Do you see any way to make it easier for others to run all tests / examples automatically?

We completely change the test files. Now everyone can run tests smoothly. If you try to run a test without GCS credentials, the specific test will simply be skipped. All the necessary parameters to run tests are set in testthat.R file. You can also run R CMD check with the argument --run-dontrun with no problems. drive credentials are necessary.

> R CMD check --as-cran reports some documentation issues of print.ee.computedobject.ComputedObject(): "Found the following apparent S3 methods exported but not registered: print.ee.computedobject.ComputedObject"

Thanks, it was fixed.

**Comment by appelmar** [comment]
Great, thanks for addressing the issues, well done!
